src/dhtproto/node/neo/request/Remove.d

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+
+    Remove request protocol.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module dhtproto.node.neo.request.Remove;
+
+import swarm.neo.node.IRequestHandler;
+
+/*******************************************************************************
+
+    v0 Remove request protocol.
+
+*******************************************************************************/
+
+public abstract class RemoveProtocol_v0 : IRequestHandler
+{
+    import swarm.neo.node.RequestOnConn;
+    import dhtproto.common.Remove;
+    import dhtproto.node.neo.request.core.Mixins;
+
+    import ocean.transition;
+
+    /// Mixin the initialiser and the connection and resources members.
+    mixin IRequestHandlerRequestCore!();
+
+    /// Response status code to send to client.
+    private MessageType response;
+
+    /***************************************************************************
+
+        Called by the connection handler immediately after the request code and
+        version have been parsed from a message received over the connection.
+        Allows the request handler to process the remainder of the incoming
+        message, before the connection handler sends the supported code back to
+        the client.
+
+        Note: the initial payload is a slice of the connection's read buffer.
+        This means that when the request-on-conn fiber suspends, the contents of
+        the buffer (hence the slice) may change. It is thus *absolutely
+        essential* that this method does not suspend the fiber. (This precludes
+        all I/O operations on the connection.)
+
+        Params:
+            init_payload = initial message payload read from the connection
+
+    ***************************************************************************/
+
+    public void preSupportedCodeSent ( Const!(void)[] init_payload )
+    {
+        auto parser = this.connection.event_dispatcher.message_parser();
+
+        auto channel = parser.getArray!(char)(init_payload);
+        auto key = *parser.getValue!(hash_t)(init_payload);
```

Comment:
For this case, where all fields in the message body are known at compile time, `parseBody` is intended to be used:
```d
mstring channel;
hash_t key;
parser.parseBody(init_payload, channel, key);
```
`parseBody` does an additional check if the end of `init_payload` is reached after these fields and throws a "message too long" exception if there are more data.

Reply:
Updated.